src/progressive-profiling/ProgressiveProfiling.jsx

```diff
@@ -90,17 +91,18 @@ const ProgressiveProfiling = (props) => {
 
   useEffect(() => {
     if (registrationEmbedded) {
-      getFieldDataFromBackend({ is_welcome_page: true });
+      getFieldDataFromBackend({ is_welcome_page: true, next: queryParams?.next });
     }
-  }, [registrationEmbedded, getFieldDataFromBackend]);
+  }, [registrationEmbedded, getFieldDataFromBackend, queryParams?.next]);
```

Comment:
We can use queryParams directly
```suggestion
  }, [registrationEmbedded, getFieldDataFromBackend, queryParams]);
```

Reply:
We can't use objects directly in dependency. It causes the infinite useEffect loop due to memory address change.

Reply:
@mubbsharanwar the useEffect should not run infinite when having object as dependency, it must be some other reason that useEffect is running infinitely for queryParams.

Reply:
Yes. The queryParams are constant and will only change once

Reply:
Why should not we use objects as a dependency? https://dmitripavlutin.com/react-useeffect-infinite-loop/#21-avoid-objects-as-dependencies